diff --git a/cf/src/connection.js b/cf/src/connection.js
index 8e79170aeb13efe8c3c77ccede0cd3a115b5e1b5..e8ac8871964302ecf70c4127baf074f67f490a30 100644
--- a/cf/src/connection.js
+++ b/cf/src/connection.js
@@ -427,6 +427,7 @@ function Connection(options, queues = {}, { onopen = noop, onend = noop, onclose
       error(Errors.connection('CONNECTION_DESTROYED', options))
 
     clearImmediate(nextWriteTimer)
+    chunk = nextWriteTimer = null
     if (socket) {
       socket.removeListener('data', data)
       socket.removeListener('connect', connected)
@@ -440,6 +441,7 @@ function Connection(options, queues = {}, { onopen = noop, onend = noop, onclose
     remaining = 0
     incomings = null
     clearImmediate(nextWriteTimer)
+    chunk = nextWriteTimer = null
     socket.removeListener('data', data)
     socket.removeListener('connect', connected)
     idleTimer.cancel()
diff --git a/cjs/src/connection.js b/cjs/src/connection.js
index 07f6716702ac888215c86ad6e0071aaf55ae519c..16536624d7075b7707f2c72c321f37a29af18e4d 100644
--- a/cjs/src/connection.js
+++ b/cjs/src/connection.js
@@ -425,6 +425,7 @@ function Connection(options, queues = {}, { onopen = noop, onend = noop, onclose
       error(Errors.connection('CONNECTION_DESTROYED', options))
 
     clearImmediate(nextWriteTimer)
+    chunk = nextWriteTimer = null
     if (socket) {
       socket.removeListener('data', data)
       socket.removeListener('connect', connected)
@@ -438,6 +439,7 @@ function Connection(options, queues = {}, { onopen = noop, onend = noop, onclose
     remaining = 0
     incomings = null
     clearImmediate(nextWriteTimer)
+    chunk = nextWriteTimer = null
     socket.removeListener('data', data)
     socket.removeListener('connect', connected)
     idleTimer.cancel()
diff --git a/src/connection.js b/src/connection.js
index 1b1cccde43b4570d5d071d6ffaab2669b3c2065a..54469592072a071572641d64ddb17779988f697f 100644
--- a/src/connection.js
+++ b/src/connection.js
@@ -425,6 +425,7 @@ function Connection(options, queues = {}, { onopen = noop, onend = noop, onclose
       error(Errors.connection('CONNECTION_DESTROYED', options))
 
     clearImmediate(nextWriteTimer)
+    chunk = nextWriteTimer = null
     if (socket) {
       socket.removeListener('data', data)
       socket.removeListener('connect', connected)
@@ -438,6 +439,7 @@ function Connection(options, queues = {}, { onopen = noop, onend = noop, onclose
     remaining = 0
     incomings = null
     clearImmediate(nextWriteTimer)
+    chunk = nextWriteTimer = null
     socket.removeListener('data', data)
     socket.removeListener('connect', connected)
     idleTimer.cancel()
